cmds/dhclient/dhclient.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"time"
+
+	"github.com/d2g/dhcp4"
+	"github.com/d2g/dhcp4client"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	defaultIface = "eth0"
+	// slop is the slop in our lease time.
+	slop = 10
+)
+
+var (
+	iList        = []string{defaultIface}
+	renewals     = flag.Int("renewals", 2, "Number of DHCP renewals before exiting")
+	verbose      = flag.Bool("verbose", false, "Verbose output")
+	debug        = func(string, ...interface{}) {}
+	leasetimeout = flag.Int("timeout", 600, "Lease timeout in seconds")
+)
+
+/*
+ * Example Client
+ */
+func dhclient(ifname string, timeout time.Duration, done chan error) {
+	var err error
+
+	// if timeout is < 10 seconds, let's get real.
+	if timeout < slop {
+		timeout = 2 * slop
+	}
+
+	n, err := ioutil.ReadFile(fmt.Sprintf("/sys/class/net/%s/address", ifname))
+	if err != nil {
+		done <- fmt.Errorf("Can't get mac for %v", ifname)
+		return
+	}
+	// This is truly amazing but /sys appends newlines to all this data.
+	n = n[:len(n)-1]
```

Comment:
I would suggest you do:
n = strings.TrimSpace(n)

If they ever break that habit or add another whitespace character it's going to break this code. This makes it a tiny bit more robust.